### PR TITLE
(PSKD-434) AWS no longer has a default storage class with K8s 1.30

### PR DESCRIPTION
**Pull request by dhoucgitter**
### Changes required:
From https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions-standard.html
> Starting with 1.30, Amazon EKS no longer includes the default annotation on the gp2 StorageClass resource applied to newly created clusters. This has no impact if you are referencing this storage class by name. You must take action if you were relying on having a default StorageClass in the cluster.

The changes made here are helping to mitigate issues arising from the last sentence above.

### Testing:
Internal testing of the proposed changes have been positive.

### CODE REVIEW

**Review by saschjmil — APPROVED** [review]

File: main.tf [hunk 1]
```diff
@@ -225,6 +225,17 @@ module "kubeconfig" {
   depends_on = [module.eks.cluster_name] # The name/id of the EKS cluster. Will block on cluster creation until the cluster is really ready.
 }
 
+# Normally, the use of local-exec below is avoided. It is used here to patch the gp2 storage class as the default storage class for EKS 1.30 and later clusters.
+# In the near future, adopting a newer version of the aws-ebs-csi-driver will create a new gp3-based storage class which will become the default storage class.
```
Comment (undr-rowr) [inline main.tf:229]: Can we go ahead and create that ticket, then, add to this explanation for posterity?

Comment (dhoucgitter) [reply main.tf:229]: Sure, @undr-rowr, will be pushing up a new commit with that ticket reference shortly.

Comment (dhoucgitter) [reply main.tf]: @undr-rowr, s/b ready for re-review

**Review by joboon — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (dhoucgitter) [reply main.tf]: Done, see the latest commit for change.

**Pull request merged**